### ISSUE TITLE
Unwanted extra features in dfm?

### ISSUE THREAD

**Issue by luketudge** [elizagrames/litsearchr#40]
I see that `create_dfm()` does some stemming and partial matching, which is very useful. But in some cases it is overzealous and adds in new unwanted features. Here is an example:

```r
dfm <- create_dfm(
  elements = c(
    "Black-backed woodpecker occupancy in burned and beetle-killed forests",
    "Burnt and black-backed woodpeckerless forests: A sad prospect",
    "Can black-backed woodpeckers get sunburn?"
  ),
  features = c("black-backed woodpecker", "burn")
)

print(dfm$dimnames$Terms)
```
```
[1] "black-backed woodpecker" "burned" "black-backed woodpeckerless" "burnt" "sunburn"
```

In addition to the requested terms, we get any term that *contains* a requested term. This might be desirable in a lot of cases, but even when it is, any derived terms that are matched this way should probably not appear as new terms, but instead be counted as instances of the root term.

**Comment by elizagrames** [comment]
Thanks for finding this problem, @luketudge! Sorry to take forever to get around to this, but I believe this is now fixed by  [44d3252](https://github.com/elizagrames/litsearchr/commit/44d32522b529d0976cbd12a6411394b45fa286d3) which cleans up the code that was created when merging with synthesisr.

**Comment by luketudge** [comment]
Thanks! Nicely fixed. And neater too, since the feature columns of the DFM are now simply those requested with the `features` argument to `create_dfm()`.

```R
dfm <- create_dfm(
    elements = c(
        "Black-backed woodpecker occupancy in burned and beetle-killed forests",
        "Burnt and black-backed woodpeckerless forests: A sad prospect",
        "Can black-backed woodpeckers get sunburn?"
    ),
    features = c("black-backed woodpecker", "burn")
)
colnames(dfm)
```
```
[1] "black-backed woodpecker" "burn"
```
